nexus/types/src/deployment.rs

```diff
@@ -828,3 +794,484 @@ pub struct UnstableReconfiguratorState {
+            self.builder.push_record(vec![subheading]);
+        }
+
+        /// Push a zone record onto the table.
+        ///
+        /// The first column is the indent and prefix column, and the rest of
+        /// the columns are the zone record.
+        fn push_zone_record(
+            &mut self,
+            first_column: String,
+            zone: &BlueprintZoneConfig,
+        ) {
+            self.builder.push_record(vec![
+                first_column,
+                zone.config.id.to_string(),
+                zone.zone_policy.to_string(),
+                zone.config.zone_type.tag().to_string(),
+                zone.config.underlay_address.to_string(),
+            ]);
+        }
+
+        /// Push a change table onto the table.
+        ///
+        /// For diffs, this contains a table of changes between two zone
+        /// records.
+        fn push_change_table(
+            &mut self,
+            before: &BlueprintZoneConfig,
+            after: &BlueprintZoneConfig,
+        ) {
+            self.change_tables.push(self.builder.count_records());
+            let table = make_zone_change_table(before, after);
+            self.builder.push_record(vec![
+                format!("{CHANGED_PREFIX}{ZONE_INDENT}"),
+                before.config.id.to_string(),
+                table.to_string(),
+            ]);
+        }
+
+        fn build(self) -> Table {
+            let mut table = self.builder.build();
+            apply_general_settings(&mut table);
+            apply_heading_settings(&mut table, &self.headings);
+            apply_subheading_settings(&mut table, &self.subheadings);
+            apply_change_table_settings(&mut table, &self.change_tables);
+
+            table
+        }
+    }
+
+    fn make_zone_change_table(
+        before: &BlueprintZoneConfig,
+        after: &BlueprintZoneConfig,
+    ) -> Table {
+        let mut builder = Builder::new();
+
+        if before.zone_policy != after.zone_policy {
+            builder.push_record(vec![
+                POLICY_HEADER.to_string(),
+                before.zone_policy.to_string(),
+                ARROW.to_string(),
+                after.zone_policy.to_string(),
+            ]);
+        } else {
+            builder.push_record(vec![
+                POLICY_HEADER.to_string(),
+                before.zone_policy.to_string(),
+                ARROW_UNCHANGED.to_string(),
+                UNCHANGED.to_string(),
+            ]);
+        }
+
+        if before.config.zone_type != after.config.zone_type {
+            // There are two cases here:
+            // 1. The zone type changed (we don't expect this to happen)
+            // 2. The zone type did not change, but some of the configuration
+            //    parameters inside changed. We need to add additional text to
+            //    make that clearer.
+            //
+            // TODO: also print out the exact bits of configuration that
```

Comment:
Do you want to do this on this PR, in a followup, or file an issue for someone to pick up later? (I have no preference between these - the old format also didn't show the details here, right?)

Reply:
Definitely in a follow up, will file an issue for this. A question that immediately occurs to me is: why are diffs special, why not display all the configs for everything? Maybe we should have an extended mode in the displayer to make that possible.